### ISSUE TITLE
Problem with building large FM index

### ISSUE THREAD

**Issue by tloka** [seqan/seqan3#2759]
## Platform

  - SeqAn version: 3.1.0-rc.1 (b5b352625)
  - Operating system: Ubuntu 20.04.2 LTS
  - Compiler: gcc 9.3.0

## Question
I have some trouble when trying to build a large FM index.  
I am using the FM index in multi-sequence mode and `dna_5`; underlying SDSL parameters are defaulted:
```c++
seqan3::fm_index<seqan3::dna5, seqan3::text_layout::collection>
```

My code works fine when building an index including 2.018 sequences of total size ~1.2GB.   
  
However, I first extended my database to 53.502 sequences of total size ~4.8GB.  
In this setup, I get the following error:  
```
malloc(): invalid size (unsorted)
```
The error is thrown from `sdsl::memory_manager::resize<...>`, the last call from SeqAn3 was here:
```c++
<fm_index.hpp; line 315>
text_begin = sdsl::sd_vector<>(builder);
```
  
Then, to check whether there is something wrong with the new sequences, I just tried to build an FM index from twice the initial database (i.e., two copies per sequence). I am getting the same error there, so for me it looks like some memory handling is not properly working for the larger database with my current setup.  
  
I am currently running the index build on a machine with 500GB RAM, which should be more than sufficient for the given tasks. I also tried on different machine with 16GB RAM plus 80GB swap, producing the same error. Building the 1.2GB index consumes approx. 12GB (should it actually be that high?).  
I would be happy about any hint whether (and what) I need to adapt to enable building of such large indexes (do I need a special config for the SDSL wt?). Sorry if I missed something in the documentation again ..  
  
PS: For some setup with other large datasets I tried, instead of the error resulting from the line mentioned above, I get a segmentation fault resulting from
```c++
<fm_index.hpp; line 324>
std::ranges::move(text | ...)
```
leading to a segmentation fault in some `addressof()` call of `std::move`.

**Comment by eseiler** [comment]
Should be fixed by https://github.com/seqan/seqan3/pull/2756

**Comment by tloka** [comment]
> 
> 
> Should be fixed by #2756

Thanks! Sorry, I scrolled through the issues to see if there is something related, I didn't think about PRs...

**Comment by tloka** [comment]
> 
> 
> Did you locally try whether it indeed fixes the problem?

Yes, it does.

**Comment by eseiler** [comment]
Fixed by #2756

**Comment by tloka** [comment]
Thanks for the quick fix!
